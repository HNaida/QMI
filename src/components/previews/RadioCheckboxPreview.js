import React from "react";
import {Checkbox, Grid, Radio, Typography} from "@material-ui/core";

const RadioCheckboxPreview = ({question}) => {
    return (
        <Grid container direction="column" spacing={1}>
            {question.options.map((option, index) => (
                <Grid item xs key={option + index}>
                    <Grid
                        container
                        direction="row"
                        justify="flex-start"
                        alignItems="center"
                    >
                        <Grid
                            item
                            xs={2}
                            style={{
                                textAlign: "center",
                            }}
                        >
                            {question.type === "checkbox" ? <Checkbox/> : <Radio/>}
                        </Grid>
                        <Grid
                            item

                            xs={9}
                            style={{
                                wordWrap: "break-word",
                                textAlign: "left",
                            }}
                        >
                            <Typography variant="body1" data-cy={option}>
                                {typeof option === "string" ? option : option.title}
                            </Typography>
                        </Grid>
                    </Grid>
                </Grid>
            ))}
        </Grid>
    );
};

export default RadioCheckboxPreview;
