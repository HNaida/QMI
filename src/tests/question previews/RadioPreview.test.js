import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react'
import RadioCheckboxPreview from '../../components/previews/RadioCheckboxPreview'

configure({ adapter: new Adapter() });

describe('<RadioCheckboxPreview/>', () => {
    it('renders a radio question', () => {
        const question=  {
            id: "v12",
            type: "radio",
            title: "Hello Kitty",
            options: ["option 1", "option 22", "option 3", "option 4"],
        };
        const component = shallow(<RadioCheckboxPreview question={question}/>);
        // expect(component).toMatchSnapshot();
    });
});
