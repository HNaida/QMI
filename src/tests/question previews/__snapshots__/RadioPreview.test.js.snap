// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioCheckboxTypePreview/> renders a radio question 1`] = `ShallowWrapper {}`;
